hawkbit-dmf/hawkbit-dmf-amqp/src/test/java/org/eclipse/hawkbit/amqp/AmqpMessageHandlerServiceTest.java

```diff
@@ -176,23 +180,60 @@ public void wrongContentType() {
-        final ArgumentCaptor<URI> uriCaptor = ArgumentCaptor.forClass(URI.class);
+        targetIdCaptor = ArgumentCaptor.forClass(String.class);
+        uriCaptor = ArgumentCaptor.forClass(URI.class);
         when(controllerManagementMock.findOrRegisterTargetIfItDoesNotExist(targetIdCaptor.capture(),
                 uriCaptor.capture())).thenReturn(targetMock);
         when(controllerManagementMock.findOldestActiveActionByTarget(any())).thenReturn(Optional.empty());
 
-        amqpMessageHandlerService.onMessage(message, MessageType.THING_CREATED.name(), TENANT, VIRTUAL_HOST);
+        amqpMessageHandlerService.onMessage(createMessage(new byte[0], getThingCreatedMessageProperties(knownThingId)),
+                MessageType.THING_CREATED.name(), TENANT, VIRTUAL_HOST);
 
         // verify
         assertThat(targetIdCaptor.getValue()).as("Thing id is wrong").isEqualTo(knownThingId);
-        assertThat(uriCaptor.getValue().toString()).as("Uri is not right").isEqualTo("amqp://"+VIRTUAL_HOST+"/MyTest");
+        assertThat(uriCaptor.getValue().toString()).as("Uri is not right")
+                .isEqualTo("amqp://" + VIRTUAL_HOST + "/MyTest");
+    }
+
+    @Test
+    @Description("Tests the creation of a target/thing with specified name by calling the same method that incoming RabbitMQ messages would access.")
+    public void createThingWithName() {
+        final String knownThingId = "2";
+        final DmfCreateThing targetProperties = new DmfCreateThing();
+        targetProperties.setName("NonDefaultTargetName");
+
+        final Target targetMock = mock(Target.class);
+
+        targetIdCaptor = ArgumentCaptor.forClass(String.class);
+        uriCaptor = ArgumentCaptor.forClass(URI.class);
+        final ArgumentCaptor<String> targetNameCaptor = ArgumentCaptor.forClass(String.class);
+
+        when(controllerManagementMock.findOrRegisterTargetIfItDoesNotExist(targetIdCaptor.capture(),
+                uriCaptor.capture(), targetNameCaptor.capture())).thenReturn(targetMock);
+        when(controllerManagementMock.findOldestActiveActionByTarget(any())).thenReturn(Optional.empty());
+
+        amqpMessageHandlerService.onMessage(
+                createMessage(targetProperties, getThingCreatedMessageProperties(knownThingId)),
+                MessageType.THING_CREATED.name(), TENANT, "vHost");
+
+        assertThat(targetIdCaptor.getValue()).as("Thing id is wrong").isEqualTo(knownThingId);
+        assertThat(uriCaptor.getValue().toString()).as("Uri is not right").isEqualTo("amqp://vHost/MyTest");
+        assertThat(targetNameCaptor.getValue()).as("Thing name is not right").isEqualTo(targetProperties.getName());
+    }
+
+    @Test
+    @Description("Tests not allowed body in message")
+    public void createThingWithWrongBody() {
+        final String knownThingId = "3";
+
+        assertThatExceptionOfType(AmqpRejectAndDontRequeueException.class)
```

Comment:
check if you should change exception type to `MessageConversionException`

Reply:
Changed to `MessageConversionException`.